core/network-libp2p/src/custom_proto/behaviour.rs

```diff
@@ -213,7 +213,7 @@ impl<TMessage, TSubstream> CustomProto<TMessage, TSubstream> {
 			// DisabledPendingEnable => Disabled.
 			PeerState::DisabledPendingEnable { open, connected_point, timer } => {
 				debug!(target: "sub-libp2p", "PSM <= Dropped({:?})", peer_id);
-				self.peerset.dropped(peer_id);
+				self.peerset.dropped(peer_id.clone());
```

Comment:
Just out of curiosity, why do you think it's better to clone on the call site? My preference would be to pass a reference and clone only if needed. In that case it doesn't really matter, since we know that current implementation requires ownership anyway, but IMHO passing reference is more future proof (shall the impl change).